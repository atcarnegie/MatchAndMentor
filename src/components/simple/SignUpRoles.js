import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import Button from './Button';
import ScreenHeader from './ScreenHeader';
import { submitRoles, updateMentorRole, updateMenteeRole } from '../../actions';
import { ROLES } from '../../actions/types';

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'space-between',
  },
  buttonStyle: {
    backgroundColor: '#55A838',
    margin: 20,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center',
  },
  textStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
  },
  roleStyle: {
    backgroundColor: '#e9e9e9',
    marginLeft: 20,
    marginRight: 20,
    marginBottom: 20,
    height: 45,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingRight: 20,
  },
  roleTextStyle: {
    color: 'black',
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
    marginLeft: 20,
  },
};

function submit(props) {
  if (props.mentorSelected && props.menteeSelected) {
    props.dispatch(submitRoles(ROLES.MENTOR_AND_MENTEE)); // both mentor and mentee
  } else if (props.mentorSelected) {
    props.dispatch(submitRoles(ROLES.MENTOR)); // mentor only
  } else {
    props.dispatch(submitRoles(ROLES.MENTEE)); // mentee only
  }
}

function SignUpRoles(props) {
  return (
    <View style={styles.container}>
      <View>
        <ScreenHeader title="Your Roles" />
        <Button style={
          props.mentorSelected
            ? { ...styles.roleStyle, backgroundColor: '#55A838' }
            : styles.roleStyle
          }
          onPress={() => props.dispatch(updateMentorRole(!props.mentorSelected))}
          testID="mentor-button">
          <Text style={styles.roleTextStyle}>Mentor</Text>
          <Icon name="check-circle-o" size={30} color='black' />
        </Button>
        <Button style={
          props.menteeSelected
            ? { ...styles.roleStyle, backgroundColor: '#55A838' }
            : styles.roleStyle
          }
          onPress={() => props.dispatch(updateMenteeRole(!props.menteeSelected))}
          testID="mentee-button">
          <Text style={styles.roleTextStyle}>Mentee</Text>
          <Icon name="check-circle-o" size={30} color='black' />
        </Button>
      </View>
      <View>
        <Button
          style={
            (props.mentorSelected || props.menteeSelected)
              ? styles.buttonStyle
              : { ...styles.buttonStyle, backgroundColor: '#e9e9e9' }
          }
          onPress={() => submit(props)}
          disabled={!props.mentorSelected && !props.menteeSelected}
          testID="roles-submit-button">
          <Text style={styles.textStyle}>Continue</Text>
        </Button>
      </View>
    </View>
  );
}

export default SignUpRoles;
