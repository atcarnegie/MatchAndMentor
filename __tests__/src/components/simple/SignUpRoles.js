/* eslint-env jest */
'use strict'
import React from 'react'
import SignUpRoles from '../../../../src/components/simple/SignUpRoles'
import renderer from 'react-test-renderer'
import { findById } from '../../../../setupTests'
import firebase from 'firebase'

firebase.initializeApp({
  apiKey: 'AIzaSyAsjC_abpWmFCYmNoQl257nc807hHE_gsY',
  authDomain: 'matchandmentorapp.firebaseapp.com',
  databaseURL: 'https://matchandmentorapp.firebaseio.com',
  projectId: 'matchandmentorapp',
  storageBucket: 'matchandmentorapp.appspot.com',
  messagingSenderId: '1068797055216'
})

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

it('renders correctly', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={false} menteeSelected={false} />)
  expect(tree).toMatchSnapshot()
})

it('calls submit on button press with mentor selected', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={true} menteeSelected={false} />)
  const submitButton = findById(tree.toJSON(), 'roles-submit-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls submit on button press with mentee selected', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={false} menteeSelected={true} />)
  const submitButton = findById(tree.toJSON(), 'roles-submit-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls submit on button press with mentor and mentee selected', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={true} menteeSelected={true} />)
  const submitButton = findById(tree.toJSON(), 'roles-submit-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls updateMentorRole on button press', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={true} menteeSelected={false} />)
  const submitButton = findById(tree.toJSON(), 'mentor-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})

it('calls updateMenteeRole on button press', () => {
  const tree = renderer.create(<SignUpRoles dispatch={jest.fn()} title='Your Role' roles={null} mentorSelected={false} menteeSelected={true} />)
  const submitButton = findById(tree.toJSON(), 'mentee-button')
  submitButton.props.onPress()
  expect(tree.toTree().props.dispatch).toBeCalled()
})
