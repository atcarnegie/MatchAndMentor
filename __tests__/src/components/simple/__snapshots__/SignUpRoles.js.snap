// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 28,
          }
        }
      >
        Your Roles
      </Text>
    </View>
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={undefined}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e9e9e9",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "paddingRight": 20,
        }
      }
      testID="mentor-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
            "marginLeft": 20,
          }
        }
      >
        Mentor
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </TouchableHighlight>
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={undefined}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#e9e9e9",
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "marginBottom": 20,
          "marginLeft": 20,
          "marginRight": 20,
          "paddingRight": 20,
        }
      }
      testID="mentee-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
            "marginLeft": 20,
          }
        }
      >
        Mentee
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </TouchableHighlight>
  </View>
  <View>
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={true}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#e9e9e9",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="roles-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
